### PR TITLE
feat: Add mute category methods

### PR DESCRIPTION
**Pull request by borismelnik**
Add muting category methods

Important changes:
- [x] Something worth noting for reviewers.

Closes #https://github.com/status-im/status-desktop/issues/5975

### CODE REVIEW

File: protocol/messenger_communities.go [hunk 1]
```diff
@@ -347,6 +347,26 @@ func (m *Messenger) SetMuted(communityID types.HexBytes, muted bool) error {
 	return m.communitiesManager.SetMuted(communityID, muted)
 }
 
+func (m *Messenger) MuteCategory(communityID string, categoryID string, muted bool) error {
```
Comment (cammellos) [inline protocol/messenger_communities.go]: Maybe I would rename this method as something like `SetMutePropertyOnChats` or something similar (can't figure out a good name :) ), `MuteCategory` gave me the impression that you want to mute the chats, but you might effectively unmuting them.

**Review by cammellos — APPROVED** [review]
Looks good, only a comment about naming, a test would be nice :)

File: protocol/messenger_communities.go (hunk 1 above)
Comment (Samyoul) [inline protocol/messenger_communities.go]: `categoryID` is passed in as a parameter but it isn't used in this function.

How does this function mute a category if it doesn't target a category?

Comment (borismelnik) [reply protocol/messenger_communities.go]: @cammellos agreed :) will fix

Comment (borismelnik) [reply protocol/messenger_communities.go]: fixed :)

**Review by 0x-r4bbit — APPROVED** [review]
LGTM!

As @cammellos said, a 1-2 tests would be wonderful. Let me know if you need help with this!

File: protocol/communities/community_categories.go [hunk 2]
```diff
@@ -6,6 +6,17 @@ import (
...
+func (o *Community) ChatsByCategoryID(categoryID string) []string {
+	var chatIDs []string
+	for chatID, chat := range o.config.CommunityDescription.Chats {
+		if chat.CategoryId == categoryID {
+			chatIDs = append(chatIDs, chatID)
+		}
+	}
+
+	return chatIDs
+}
```
Comment (richard-ramos) [inline protocol/communities/community_categories.go]:
```suggestion
	var chatIDs []string
	if o == nil || o.config == nil || o.config.CommunityDescription == nil {
		return chatIDs 
	}
	
	for chatID, chat := range o.config.CommunityDescription.Chats {
		if chat.CategoryId == categoryID {
			chatIDs = append(chatIDs, chatID)
		}
	}

	return chatIDs
}
```

File: protocol/communities/community_categories.go [hunk 3]
```diff
@@ -6,6 +6,17 @@ import (
 	"github.com/status-im/status-go/protocol/protobuf"
 )
 
+func (o *Community) ChatsByCategoryID(categoryID string) []string {
```
Comment (richard-ramos) [inline protocol/communities/community_categories.go:9]:
```suggestion
func (o *Community) ChatsByCategoryID(categoryID string) []string {
	o.mutex.Lock()
	defer o.mutex.Unlock()
```

**Comment by anastasiyaig** [comment]
@borismelnik tests failed, please take a look

File: protocol/communities/community_categories.go (hunk 3 above)
Comment (borismelnik) [reply protocol/communities/community_categories.go:9]: done :)

File: protocol/communities/community_categories.go (hunk 2 above)
Comment (borismelnik) [reply protocol/communities/community_categories.go]: done :)

**Comment by 0x-r4bbit** [comment]
@borismelnik The tests still seems to be failing on CI

**Review by richard-ramos — APPROVED** [review]

**Pull request merged**